packages/config/src/validation/validate-condition.ts

```diff
@@ -1,12 +1,31 @@
 import type { Conditions } from '@pandacss/types'
 import type { AddError } from '../types'
+import { isString } from '@pandacss/shared'
 
 export const validateConditions = (conditions: Conditions | undefined, addError: AddError) => {
   if (!conditions) return
 
   Object.values(conditions).forEach((condition) => {
-    if (!condition.startsWith('@') && !condition.includes('&')) {
-      addError('conditions', `Selectors should contain the \`&\` character: \`${condition}\``)
+    if (isString(condition)) {
+      if (!condition.startsWith('@') && !condition.includes('&')) {
+        addError('conditions', `Selectors should contain the \`&\` character: \`${condition}\``)
+      }
+
+      return
+    }
+
+    condition.forEach((c) => {
+      if (!c.startsWith('@') && !c.includes('&')) {
+        addError('conditions', `Selectors should contain the \`&\` character: \`${c}\``)
+      }
+    })
+
+    const hasAtLeastOneAmperand = condition.some((c) => c.includes('&'))
+    if (!hasAtLeastOneAmperand) {
+      addError(
+        'conditions',
+        `Mixed selectors should contain the \`&\` character in at least one segment: \`${condition.join(', ')}\``,
+      )
     }
```

Comment:
A mixed condition could definitely use two "at-rule" (say @media and @supports). Maybe this is perhaps too strict?